### PR TITLE
Bluetooth: Mesh: Light Switch sample

### PR DESCRIPTION
**Pull request by trond-snekvik**
Adds the Bluetooth Mesh Light Switch sample, which can control the
Bluetooth Mesh Light sample application. The sample demonstrates the
OnOff client model.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

### CODE REVIEW

File: samples/bluetooth/mesh/light_switch/prj.conf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+CONFIG_ASSERT=y
+CONFIG_ASSERT_LEVEL=2
+CONFIG_LOG=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_LOG_PRINTK=y
+
+CONFIG_DK_LIBRARY=y
+CONFIG_BT_MESH_DK_PROV=y
+
+# Bluetooth configuration
```
Comment (joerchan) [inline samples/bluetooth/mesh/light_switch/prj.conf]: Maybe we could do something similar for BT MESH samples?
https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1594/files#diff-25775b8be1637aad0917e4de7644d3c8R7

File: samples/bluetooth/mesh/light_switch/src/model_handler.c [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+static void attention_blink(struct k_work *work)
+{
+	static int idx;
+	const u8_t pattern[] = {
+		BIT(0) | BIT(1),
+		BIT(1) | BIT(2),
+		BIT(2) | BIT(3),
+		BIT(3) | BIT(0),
+	};
+	dk_set_leds(pattern[idx++ % ARRAY_SIZE(pattern)]);
```
Comment (joerchan) [inline samples/bluetooth/mesh/light_switch/src/model_handler.c]: blank line after declarations.

File: samples/bluetooth/mesh/light_switch/src/model_handler.c [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+/**
+ * @file
+ * @brief Model handler for the light switch.
+ *
+ * Instantiates a Generic OnOff Client model for each button on the devkit, as
+ * well as the standard foundation models. Handles all application behavior
```
Comment (joerchan) [inline samples/bluetooth/mesh/light_switch/src/model_handler.c]: What are foundation models? Config and Health?

File: samples/bluetooth/mesh/light_switch/src/model_handler.c [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+	}
+}
+
+/** Configuration server definition */
+static struct bt_mesh_cfg_srv cfg_srv = {
+	.relay = IS_ENABLED(CONFIG_BT_MESH_RELAY),
+	.beacon = BT_MESH_BEACON_ENABLED,
+	.frnd = IS_ENABLED(CONFIG_BT_MESH_FRIEND),
+	.gatt_proxy = IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY),
+	.default_ttl = 7,
```
Comment (joerchan) [inline samples/bluetooth/mesh/light_switch/src/model_handler.c]: Is this 7 something we could put in a header somewhere as recommended, or default?

File: samples/bluetooth/mesh/light_switch/src/model_handler.c [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+		BIT(2) | BIT(3),
+		BIT(3) | BIT(0),
+	};
+	dk_set_leds(pattern[idx++ % ARRAY_SIZE(pattern)]);
+	k_delayed_work_submit(&attention_blink_work, 30);
+}
+
+static void attention_on(struct bt_mesh_model *mod)
+{
+	k_delayed_work_submit(&attention_blink_work, 0);
```
Comment (joerchan) [inline samples/bluetooth/mesh/light_switch/src/model_handler.c]: K_NO_WAIT

File: samples/bluetooth/mesh/light_switch/src/model_handler.c [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+{
+	static int idx;
+	const u8_t pattern[] = {
+		BIT(0) | BIT(1),
+		BIT(1) | BIT(2),
+		BIT(2) | BIT(3),
+		BIT(3) | BIT(0),
+	};
+	dk_set_leds(pattern[idx++ % ARRAY_SIZE(pattern)]);
+	k_delayed_work_submit(&attention_blink_work, 30);
```
Comment (joerchan) [inline samples/bluetooth/mesh/light_switch/src/model_handler.c]: K_MSEC(30)

File: samples/bluetooth/mesh/light_switch/prj.conf (hunk 1 above)
Comment (trond-snekvik) [reply samples/bluetooth/mesh/light_switch/prj.conf]: I have a task in my backlog for going through the config and see if I can add more `imply`s throughout. I'd rather deal with the samples as part of that task :)

**Pull request merged**